### PR TITLE
fix overhead estimates to be more accurate

### PR DESCRIPTION
**Pull request by jreynolds01**
### Description

Update the aks estimation utilities so that overhead is calculated more accurately.

### Checklist:

- [X ] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [X ] I have added tests covering my contributions.
- [X ] I have updated the documentation accordingly.

### CODE REVIEW

File: reco_utils/azureml/aks_utils.py [hunk 1]
```diff
@@ -3,52 +3,78 @@
...
     Returns:
-        replicas: Total number of replicas supported by n_cores
+        replicas: Total number of replicas supported by the configuration
     """
-    replicas = floor((1 - overhead)*n_cores/(cpu_cores_per_replica))
-    logger.info('Approximately {} replicas are supported by {} cores.'.format(replicas, n_cores))
-    return replicas
\ No newline at end of file
+    n_cores_avail = (n_cores_per_node - 0.5) * n_nodes - 4.45
+    replicas = floor(n_cores_avail / (cpu_cores_per_replica))
```
Comment (gramhagen) [inline reco_utils/azureml/aks_utils.py]: tiny nitpick, parentheses around cpu_cores_per_replica isn't needed

File: reco_utils/azureml/aks_utils.py [hunk 2]
```diff
@@ -3,52 +3,78 @@
...
+        processing_time (float): the estimated amount of time (in seconds)
+            your service call takes
         max_qp_replica (int): maximum number of concurrent queries per replica
-        target_utilization (float): proportion of CPU utilization you think is ideal
+        target_utilization (float): proportion of CPU utilization you think is
+            ideal
 
     Returns:
-        replicas: Number of estimated replicas required to support a target number of queries per second
+        replicas: Number of estimated replicas required to support a target
```
Comment (miguelgfierro) [inline reco_utils/azureml/aks_utils.py]: this would be `int: Number of estimated...`?

File: reco_utils/azureml/aks_utils.py [hunk 3]
```diff
@@ -3,52 +3,78 @@
...
+
     Args:
-        num_replicas (int): number of replicas 
+        num_replicas (int): number of replicas
         processing_time (float): the estimated amount of time (in seconds) your service call takes
         max_qp_replica (int): maximum number of concurrent queries per replica
         target_utilization (float): proportion of CPU utilization you think is ideal
 
     Returns:
         qps: queries per second supported by the number of replicas
```
Comment (miguelgfierro) [inline reco_utils/azureml/aks_utils.py]: This would be `int: queries...`?

File: reco_utils/azureml/aks_utils.py [hunk 4]
```diff
@@ -3,52 +3,78 @@
...
+        n_cores_per_node (int): Total number of cores per node within an AKS
+            cluster that you want to use
+        n_nodes (int): Number of nodes (i.e. VMs) used in the AKS cluster
+        cpu_cores_per_replica (float): Cores assigned to each replica. This
+            can be fractional and corresponds to the
+            cpu_cores argument passed to AksWebservice.deploy_configuration()
 
     Returns:
-        replicas: Total number of replicas supported by n_cores
+        replicas: Total number of replicas supported by the configuration
```
Comment (miguelgfierro) [inline reco_utils/azureml/aks_utils.py]: same as above

**Comment by gramhagen** [comment]
we already merged this to staging, so I'm going to make the small changes and move this to push into staging

**Pull request merged**